src/promise-katas-1.js

```diff
@@ -30,86 +30,136 @@ Remember, a Promise can only resolve successfully, or reject when it fails to re
+  });
+};
 
 // 4 Create a function that returns a promise which returns an object with a key of name and value of 'Bob' if it resolves succesfully
 // Example:
 // returnBob() resolves to the value { name: 'Bob' }
 
-const returnBob = () => { };
+const returnBob = () => {
+  return new Promise((resolve, reject) => {
+    resolve({ name: "Bob" });
+  });
+};
 
 // 5 Create a function that returns a promise which returns an array with 4 strings of 'eggs', 'apples', 'milk' and 'bread' if it resolves succesfully
 // Example:
 // returnList() resolves to the value ['eggs', 'apples', 'milk', 'bread']
 
-const returnList = () => { };
+const returnList = () => {
+  return new Promise((resolve, reject) => {
+    resolve(["eggs", "apples", "milk", "bread"]);
+  });
+};
 
 // 6 Create a function that returns a promise which returns "An error occurred" if it rejects
 // Example:
 // anError() rejects to "An error occurred"
 
-const anError = () => { };
+const anError = () => {
+  return new Promise((resolve, reject) => {
+    reject("An error occurred");
+  });
+};
 
 // 7 Create a function that returns a promise which returns 666 if it rejects
 // Example:
 // theNumberOfTheBeast() rejects to the value 666
-
-const theNumberOfTheBeast = () => { };
+const theNumberOfTheBeast = () => {
+  return new Promise((resolve, reject) => {
+    reject(666);
+  });
+};
 
 // 8 Create a function that returns a promise which returns an object with a key of error and value of 500 if it rejects
 // Example:
 // internalServerError() rejects to { error: 500 }
 
-const internalServerError = () => { };
+const internalServerError = () => {
+  return new Promise((resolve, reject) => {
+    reject({ error: 500 });
+  });
+};
 
-// 9 Create a function that returns a promise which resolves to "happy" if the parameter is greater than 
+// 9 Create a function that returns a promise which resolves to "happy" if the parameter is greater than
 // or equal to 1 and rejects to "sad" if the value is less than 1
 // Example:
 // happySad(1) resolves to "happy"
 // happySad(0) rejects to "sad"
 
-const happySad = () => { };
-
-// 10 Create a function that returns a promise which resolves to "Yes. Luke, I am your father." if the 
+const happySad = (x) => {
+  return new Promise((resolve, reject) => {
+    if (x > 1) {
+      resolve("happy");
+    } else {
+      reject("sad");
+    }
+  });
+};
+
+// 10 Create a function that returns a promise which resolves to "Yes. Luke, I am your father." if the
 // parameter is "Luke" and rejects to "Not your dad." if the value is anything else
 // Example:
 // amIYourFather() resolves to "Yes. Luke, I am your father."
 // amIYourFather() rejects to "Not your dad."
 
-const amIYourFather = () => { };
-
-// 11 Create a function that returns a promise. The promise should resolve to a new function that returns "My name is <input name>" 
+function amIYourFather(x) {
+  return new Promise((resolve, reject) => {
+    if (x === "Luke") {
```

Comment:
It would be better to use a more meaningful variable name here instead of `x`, which could mean literally anything. How about `name` instead?